test/waveform_backend.js

```diff
@@ -39,54 +39,52 @@ describe("WaveformBackend", () => {
     it("should call chrome.runtime.onConnect", () => {
       wb.init();
       expect(chrome.runtime.onMessage.addListener).to.be.calledWith(
-        wb.processAudio
+        wb.processRequest
       );
     });
   });
 
-  describe("processAudio()", () => {
+  describe("processRequest()", () => {
     let request;
 
     let sendResponseSpy = sinon.spy();
 
     let stubedFetch;
 
+    const buffer = new Uint8Array([0, 1, 2, 3]);
+
     beforeEach(() => {
       request = {
         contentScriptQuery: "notTheRightSender",
-        datapoints: 128,
-        length: 1024,
-        fs: 44100,
         url: "test"
       };
 
       stubedFetch = sinon.stub(window, "fetch");
-      window.fetch.returns(Promise.resolve(mockApiResponse));
-
-      sandbox.stub(window, "OfflineAudioContext");
-      window.OfflineAudioContext.callThroughWithNew().returns(sinon.spy());
+      stubedFetch.returns(Promise.resolve(mockApiResponse()));
     });
 
     afterEach(() => {
       stubedFetch.restore();
     });
 
-    it("should return without doing anything", () => {
-      wb.processAudio(request, null, sendResponseSpy);
-
-      expect(sendResponseSpy).to.be.not.called;
-      expect(window.OfflineAudioContext).to.be.not.called;
-    });
+    // TODO: 'sendResponseSpy' will never be called in this test
```

Comment:
If this will never be called, do we need to keep it around?

Reply:
once I figure out how to test code within the `then` blocks of the Fetch promise, then I can re-enable this code.

Specifically, for this code (once it's functioning correctly): if either of those functions are called, our code is broken because this is testing a branch of an if statement that should bypass this code.